library/src/main/java/com/chuckerteam/chucker/internal/support/RequestProcessor.kt

```diff
@@ -0,0 +1,66 @@
+package com.chuckerteam.chucker.internal.support
+
+import android.content.Context
+import com.chuckerteam.chucker.R
+import com.chuckerteam.chucker.api.ChuckerCollector
+import com.chuckerteam.chucker.internal.data.entity.HttpTransaction
+import okhttp3.Request
+import okio.Buffer
+import okio.IOException
+import kotlin.text.Charsets.UTF_8
+
+internal class RequestProcessor(
+    private val context: Context,
+    private val collector: ChuckerCollector,
+    private val maxContentLength: Long,
+) {
+    fun process(request: Request, transaction: HttpTransaction): Request {
+        processMetadata(request, transaction)
+        processBody(request, transaction)
+        collector.onRequestSent(transaction)
+        return request
+    }
+
+    private fun processMetadata(request: Request, transaction: HttpTransaction) {
+        transaction.apply {
+            setRequestHeaders(request.headers)
+            populateUrl(request.url)
+
+            requestDate = System.currentTimeMillis()
+            method = request.method
+            requestContentType = request.body?.contentType()?.toString()
+            requestPayloadSize = request.body?.contentLength()
+        }
+    }
+
+    private fun processBody(request: Request, transaction: HttpTransaction) {
+        val body = request.body ?: return
+
+        val isEncodingSupported = request.headers.hasSupportedContentEncoding
+        if (!isEncodingSupported) {
+            return
+        }
+
+        val limitingSource = try {
+            Buffer().apply { body.writeTo(this) }
+        } catch (e: IOException) {
+            Logger.error("Failed to read request payload", e)
+            return
+        }.uncompress(request.headers).let { LimitingSource(it, maxContentLength) }
```

Comment:
nit: Here we mix multi line lambdas with inline lambdas that is not exactly elegant. Can you add a couple of new lines?

Reply:
I want to make sure I understand you correctly because there is only one lambda. Do you mean `try/catch` expression that is mixed with a one liner lambda next to it or generally in the project?

Reply:
> because there is only one lambda

My bad. I meant _block_ and it was related to this statement specifically. Also it's a nit so I'll leave it to your final decision :)

Maybe something like:

```
}.uncompress(request.headers).let { 
    LimitingSource(it, maxContentLength) 
}
```

or even
```